src/cmdline.rs

```diff
@@ -64,6 +64,15 @@ pub struct Opt {
     #[structopt(long = "deterministic", help = "Produce a deterministic TAB file")]
     pub deterministic: bool,
 
+    #[structopt(long = "program", help = "Use a program header")]
+    pub program: bool,
```

Comment:
What is the intent behind main vs program tlv? I know we need to support both, but does it make sense to just prioritize program header (and use it in all of our tooling) and then drop support for main at some point in the future? Or should we plan to use both, where program supports footers and main is for apps without footers.

In any case, this doesn't seem like a flag we should have, the program header should be used in any case where elf2tab needs to insert a footer.

Reply:
I think your two comments are mutually exclusive. If we want to transition to always using a program header, then it can't only be when there's a footer. E.g., it should be possible to have a TBF end at the end of the application binary. Having a footer requires you use the program header, but if you don't have footers you can still have a program header.

I originally argued we should transition to program headers and move away from main headers, and you argued instead we should continue to support both, in part for kernel versioning reasons. So a TBF can have both a main header (for older kernels) and a program header (for newer kernels). You argued that if both are present it should be up to the kernel to decide which one to use.

Reply:
There are three cases I think:

- If a there is no footer:
    - We must include Main. That supports old kernels.
- If there is a footer:
    - We can include Program and not Main. Arguably don't need to support old kernels, why compile with footer for old kernels.
    - We can include both.

The only case where Program is optional is when there is no footer. In that case, I suppose elf2tab could be explicitly instructed to add a Program tlv using the `--program` flag. However, I don't think that should be an option. Either we have elf2tab always insert a Program header, or elf2tab only uses it when necessary, but we don't expose this as an option, mainly because 1) users shouldn't have to think through this logic, and 2) its confusing what should happen if a footer is specified but _not_ `--program`.

Also, I'm supportive of moving towards only Program, but for backwards compatibility we have to support both.

Reply:
OK, I will remove the option.